spring-jdbc/src/main/java/io/crate/spring/jdbc/samples/da/BlogPostJDBCTemplateDao.java

```diff
@@ -0,0 +1,147 @@
+                        + "p.\"user\"['name'] as username, p.\"user\"['location'] as userlocation, "
+                        + "p.created as created, c.name as country, c.geometry as area, "
+                        + "p.image_ref as imageRef, p.like_count as likes " + "FROM %s AS p, %s AS c "
+                        + "WHERE within(p.\"user\"['location'], c.geometry) " + "AND p.id = ?",
+                POST_TABLE, COUNTRIES_TABLE), (rs, rowNum) -> stdRowMapper.apply(rs), id);
+
+        return queryResult != null && queryResult.size() == 1 ? Optional.of(queryResult.get(0)) : Optional.empty();
+    }
+
+    public Optional<BlogPost> insertPost(@NonNull BlogPost newPost) {
+
+        if (newPost.getId() == null) {
+            newPost.setId(UUID.randomUUID().toString());
+        }
+
+        if (this.jdbcTemplate.update(
+            String.format(
+                "INSERT INTO %s " +
+                "(id, \"user\", text, image_ref, created, like_count) " +
+                "VALUES (?, ?, ?, ?, ?, ?)", POST_TABLE), 
+                newPost.getId(), this.objectsFactory.createUserObject(newPost), newPost.getText(), 
+                newPost.getImage_ref(), newPost.getCreated(), newPost.getLike_count()) != 1)
+            return Optional.empty();
+        else {
+            // Since CrateDB is 'only' eventually consistent, we'll have to refresh the table before reading
+            this.jdbcTemplate.execute(String.format("REFRESH TABLE %s", POST_TABLE));
+            return this.getPost(newPost.getId());
+        }
+    }
+
+    public Optional<BlogPost> updatePostText(@NonNull String id, String newText) {
+
+       if (this.jdbcTemplate.update(
+           String.format("UPDATE %s SET text = ? WHERE id = ?", POST_TABLE), newText, id) != 1)
+            return Optional.empty();
+        else {
+            // Since CrateDB is 'only' eventually consistent, we'll have to refresh the table before reading
+            this.jdbcTemplate.execute(String.format("REFRESH TABLE %s", POST_TABLE));
+            return this.getPost(id);
+        }
+    }
+
+    public Optional<BlogPost> incrementLikeCount(@NonNull String id) {
+
+        if (this.jdbcTemplate.update(
+            String.format("UPDATE %s SET like_count = like_count + 1 WHERE id = ?", POST_TABLE), id) != 1)
+            return Optional.empty();
+        else {
+            // Since CrateDB is 'only' eventually consistent, we'll have to refresh the table before reading
+            this.jdbcTemplate.execute(String.format("REFRESH TABLE %s", POST_TABLE));
+            return this.getPost(id);
+        }
+    }
+
+    public boolean deletePost(@NonNull String id) {
+
+        return this.jdbcTemplate.update(
+                    String.format("DELETE FROM %s", POST_TABLE) + " WHERE id = ?", id) == 1;
+    }
+
+    public List<BlogPost>searchPostsWithQuery(@NonNull final String query) {
+
+        List<BlogPost> result = this.jdbcTemplate.query((String.format(
```

Comment:
Same here, why not simply add the tablenames directly to the statement ?

Reply:
Same again. Same way as original JDBC sample

Reply:
And btw. I think it's a good idea to have the table names not part of the statement. If you want to change the table names, you'll have to change tens of statements correctly!